src/confcom/azext_confcom/tests/latest/test_confcom_kata.py

```diff
@@ -49,22 +48,37 @@ def test_invalid_input_path(self):
 
     def test_invalid_config_map_path(self):
         filename = "pod.yaml"
-        with open(filename, "w") as f:
-            f.write(KataPolicyGen.pod_string)
-        with self.assertRaises(SystemExit) as wrapped_exit:
-            katapolicygen_confcom(
-                filename, "fakepath/configmap.yaml",
-            )
-        os.remove(filename)
+        try:
+            with open(filename, "w") as f:
+                f.write(KataPolicyGen.pod_string)
+            with self.assertRaises(SystemExit) as wrapped_exit:
+                katapolicygen_confcom(
+                    filename, "fakepath/configmap.yaml",
+                )
+        finally:
+            os.remove(filename)
         self.assertNotEqual(wrapped_exit.exception.code, 0)
 
-    def test_invalid_settings(self):
+    def test_valid_settings(self):
         filename = "pod2.yaml"
-        with open(filename, "w") as f:
-            f.write(KataPolicyGen.pod_string)
+        try:
+            with open(filename, "w") as f:
+                f.write(KataPolicyGen.pod_string)
+            with self.assertRaises(SystemExit) as wrapped_exit:
+                katapolicygen_confcom(
+                    filename, None
+                )
+            with open(filename, "r") as f:
+                content = f.read()
+        finally:
+            os.remove(filename)
+        self.assertEqual(wrapped_exit.exception.code, 0, "Policy not generated successfully")
```

Comment:
does `wrapped_exit` exist outside the `with` block?

Reply:
yep! test still passes